autojump.py

```diff
@@ -9,21 +9,68 @@ def hook_init(fm):
     def update_autojump(signal):
         subprocess.Popen(["autojump", "--add", signal.new.path])
 
-    fm.signal_bind('cd', update_autojump)
+    fm.signal_bind("cd", update_autojump)
     HOOK_INIT_OLD(fm)
 
-
 ranger.api.hook_init = hook_init
 
 
 class j(Command):
-    """:j
+    """:j [<text>]
+
     Uses autojump to set the current directory.
+
+    It works in the same way as the `j` function provided by autojump. In
+    addition, if fzf is present on the system, it is used to browse the
+    completion proposals returned by autojump.
     """
 
     def execute(self):
-        directory = subprocess.check_output(["autojump", self.arg(1)])
-        directory = directory.decode("utf-8", "ignore")
-        directory = directory.rstrip('\n')
-        self.fm.execute_console("cd -r " + directory)
+        import os.path
+        import re
+        from ranger.ext.get_executables import get_executables
+
+        if "autojump" not in get_executables():
+            return self.fm.notify("Could not find autojump in the PATH.", bad=True)
+
+        if "fzf" not in get_executables():
+            directory = subprocess.check_output(["autojump", self.arg(1)])
+            directory = directory.decode("utf-8", "ignore")
+            directory = directory.rstrip("\n")
+            self.fm.cd(directory)
```

Comment:
Can we do an early return to save some indentation?

Reply:
I'm not sure, if fzf is absent, an action must still be performed.